pkg/skaffold/runner/runner.go

```diff
@@ -269,7 +272,18 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*v1
 
 		if err := watcher.Register(
 			func() ([]string, error) { return dependenciesForArtifact(artifact) },
-			func() { changed.Add(artifact) },
+			func(e watch.Events) error {
+				sync, err := r.shouldSync(artifact.ImageName, artifact.Sync, e)
+				if err != nil {
+					return errors.Wrap(err, "checking sync files")
+				}
+				if !sync {
+					changed.Add(artifact)
+				} else {
+					color.Default.Fprintln(out, "Synced:", "copied", append(e.Added, e.Modified...), "deleted", e.Deleted)
```

Comment:
I *think* this is the case, but just to be sure: if we get here, is it always true that we've performed a sync?